File: Var10CSharp/Var10Cshar/SumOfDigits.cs
/*class Program
{
    // Функция для вычисления суммы цифр числа
    static int SumOfDigits(int num)
    {
        int sum = 0;
        // Если число отрицательное, меняем знак
        if (num < 0)
            num *= -1;
        while (num != 0)
        {
            sum += num % 10; // Добавляем к сумме последнюю цифру числа
            num /= 10; // Удаляем последнюю цифру числа
        }
        return sum; // Возвращаем сумму цифр числа
    }

    static void Main(string[] args)
    {
        Console.Write("Enter the number of numbers: "); // Приглашение к вводу количества чисел
        int n = int.Parse(Console.ReadLine()); // Ввод количества чисел

        Console.Write("Enter numbers separated by space: "); // Приглашение к вводу чисел через пробел
        string[] numbersStr = Console.ReadLine().Split(' '); // Ввод чисел и разделение строки по пробелам
        int[] numbers = Array.ConvertAll(numbersStr, int.Parse); // Преобразование строковых чисел в массив целых чисел

        foreach (int number in numbers)
        {
            // Проверяем, что сумма цифр числа состоит из одной цифры и число не меньше -9
            if (SumOfDigits(number) < 10 && number > -10)
            {
                Console.Write(number + " "); // Выводим число
            }
        }
        Console.WriteLine(); // Переход на новую строку
    }
}*/

File: Var10Rust/src/SumOfDigits.rs
use std::io;

// Функция для вычисления суммы цифр числа
fn sum_of_digits(mut num: i32) -> i32 {
    let mut sum = 0;
    // Если число отрицательное, меняем знак
    if num < 0 {
        num *= -1;
    }
    while num != 0 {
        sum += num % 10;
        num /= 10;
    }
    sum
}

fn main() {
    println!("Enter the number of numbers:");
    let mut input = String::new();
    io::stdin().read_line(&mut input).expect("Failed to read line");
    let n: i32 = input.trim().parse().expect("Please enter a valid number");

    println!("Enter numbers separated by space:");
    input.clear();
    io::stdin().read_line(&mut input).expect("Failed to read line");
    let numbers: Vec<i32> = input
        .split_whitespace()
        .map(|s| s.parse().expect("Please enter valid numbers"))
        .collect();

    for number in numbers.iter() {
        // Проверяем, что сумма цифр числа состоит из одной цифры
        if sum_of_digits(*number) < 10 && *number > -10 {
            print!("{} ", number);
        }
    }
    println!();
}
